pkg/securitypolicy/securitypolicyenforcer_rego.go

```diff
@@ -333,36 +325,176 @@ func (policy *regoEnforcer) denyWithError(ctx context.Context, policyError error
 	log.G(ctx).WithField("error", string(decisionJSON)).Debug("Policy decision")
 
 	base64EncodedDecisionJSON := base64.RawURLEncoding.EncodeToString(decisionJSON)
+	errorMessage := fmt.Errorf(policyDecisionPattern, base64EncodedDecisionJSON)
+	if policy.maxErrorMessageLength == 0 {
+		// indicates no message truncation
+		return fmt.Errorf(policyDecisionPattern, base64EncodedDecisionJSON)
+	}
+
+	truncated := []string{}
+	for len(errorMessage.Error()) > policy.maxErrorMessageLength {
+		actionTaken := false
+		if rawReason, ok := decision["reason"]; ok {
+			// check if it is a framework reason object
+			if reason, ok := rawReason.(rpi.RegoQueryResult); ok {
+				// check if we can remove error_objects
+				if _, ok := reason["error_objects"]; ok {
```

Comment:
we'll always reach here regardless if we need to remove `error_objects` or not. anything else we could potentially want to truncate in the future? we could potentially define a "truncateFunc" and iterate/call the funcs in a loop or something instead of potentially adding more `if !actionTaken` statements. but it's  probably ok for now...

Reply:
I like the idea of a truncate function, that's a cleaner way of doing this.

Reply:
Latest version uses this approach, I like it a lot better